src/muse/agents/factories.py

```diff
@@ -108,8 +113,18 @@ def create_newcapa_agent(
     existing = capacity.interp(year=year, method=interpolation) > 0
     assert set(existing.dims) == {"asset"}
     years = [capacity.year.min().values, capacity.year.max().values]
+
     assets = xr.Dataset()
-    assets["capacity"] = xr.zeros_like(capacity.sel(asset=existing.values, year=years))
+    if retrofit_present:
+        assets["capacity"] = xr.zeros_like(
+            capacity.sel(asset=existing.values, year=years)
+        )
+    else:
+        share = "agent_share_2"
+        technologies = kwargs["technologies"]
+        assets["capacity"] = _shared_capacity(
+            technologies, capacity, region, share, year, interpolation=interpolation
+        )
 
```

Comment:
This line forces to bring back the agent_share_2 line. It may need to be changed into smething more flexible, what if we have more than 1 agent?

Reply:
To run the model, I had to change on line 123, "2" to "1". 
However, we may follow the same approach used for the standard_agent, here

`https://github.com/SGIModel/MUSE_OS/blob/9bf1957c28b0030a442f3bc24353f10943bc64d6/src/muse/agents/factories.py#L12`